src/app/shared/courses/courses.service.ts

```diff
@@ -13,19 +13,25 @@ export class CoursesService {
         return this.courses.slice(0, page * displayLimit);
     }
 
+    getCoursesLength(): number {
+        return this.courses.length;
+    }
+
     getOrderedCourse(page: number, displayLimit: number): Course[] {
         return this.orderByName.transform(this.getList(page, displayLimit));
     }
 
     createCourse(title: string, duration: number, description: string, topRated: boolean): Course {
-        return {
-            id: Math.floor(Math.random() * 1000) + 1,
+        const course = {
+            id: this.getCoursesLength() + 1,
             title,
             dateCreation: new Date(),
             duration,
             description,
             topRated
         };
+        this.courses.push(course);
+        return course;
     }
 
     updateCourse(id: number, title: string, duration: number, description: string, topRated: boolean): Course {
```

Comment:
just `create` as it is obvious that within `CoursesService` you are managing courses.

Reply:
I suggest to use this method with different api endpoin